pkg/controllers/cluster/cluster_controller.go

```diff
@@ -111,10 +111,11 @@ func (r *RemoteClusterReconciler) EnsureCreated(ctx context.Context, resource li
 		return ctrl.Result{}, lifecycle.Failed, err
 	}
 
+	cluster.SetDefaultTokenValidityIfNeeded()
 	var tokenRequest = &tokenHelper{
 		Client:                             r.Client,
-		RemoteClusterBearerTokenValidity:   r.RemoteClusterBearerTokenValidity,
-		RenewRemoteClusterBearerTokenAfter: r.RenewRemoteClusterBearerTokenAfter,
+		RemoteClusterBearerTokenValidity:   cluster.Spec.KubeConfig.MaxTokenValidity.Duration,
+		RenewRemoteClusterBearerTokenAfter: cluster.Spec.KubeConfig.MaxTokenValidity.Duration - (24 * time.Hour),
```

Comment:
This could break because we have no minimum value for the MaxTokenValidity.
I would propose 24h minimum & default to be 72h.
Then we can keep the `r.RenewRemoteClusterBearerTokenAfter`

Reply:
Done, I set minimum value.